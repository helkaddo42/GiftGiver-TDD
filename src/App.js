import './App.css';
import React, {Component} from 'react'
import {Button} from 'react-bootstrap'
import Gifts from './Gifts'

class App extends Component {

  state={
    gifts: [],
  }

  addGift=()=>{
    const {gifts} =this.state;
    const id = gifts.map(gifts => gifts.id);
    const max_id = id.length > 0 ? Math.max(...id) : 0
    gifts.push({id : max_id+1})
    this.setState(gifts)
  }

  removeGifts=(id)=>{
    const giftsDelete = this.state.gifts.filter(idGifts => idGifts.id !== id)
    this.setState({gifts:giftsDelete})
  }

  render(){
    const {gifts} = this.state;
    const  divChildren = gifts.map(element =>{
      return(
            <Gifts 
              removeGifts={this.removeGifts}
              key={element.id}
              gifts={element.id}/>
      )
    })

    return(
      <div className="App">
          <h2>Gift Giver</h2>
        <div className='gifts-list'>
          {divChildren}
        </div>
        <Button className='btn-add' onClick={this.addGift}>Add gift</Button>
         
  
      </div>
    );
    
  }

   

  
}

export default App;
