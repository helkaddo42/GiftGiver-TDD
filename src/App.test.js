import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('APP',()=>{
  const wrapper = shallow(<App />)

  it('app component is here', ()=>{
    expect(wrapper).toMatchSnapshot()
  });

  it('we have state array gift',()=>{
    expect(wrapper.state().gifts).toEqual([])
  });

  describe('when the btn ad-gifts clicked',()=>{
   
    beforeEach(()=>{
      wrapper.find('.btn-add').simulate('click');
    })

    afterEach(()=>{
      wrapper.setState({gifts: []})
    })
    it('when click btn add new gift in state array Gifts',()=>{
      expect(wrapper.state().gifts).toEqual([{id : 1}]);
    });
  
    it('add gifts and display this gifts in dom ',()=>{
      expect(wrapper.find('.gifts-list').children().length).toEqual(1)
    })

    it('create a gifts component',()=>{
      expect(wrapper.find('Gifts').exists()).toBe(true)
    })
  })

describe('remove a gifts',()=>{
  
  const id = 1;
  beforeEach(()=>{
    wrapper.instance().removeGifts(id);
  });

  it('remove this gifts from state',()=>{
    expect(wrapper.state().gifts).toEqual([])
  })
})


})