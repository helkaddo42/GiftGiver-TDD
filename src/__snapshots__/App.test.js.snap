// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP app component is here 1`] = `ShallowWrapper {}`;
